### PR TITLE
Compatibility update for PyTorch API v1.0

### PR DESCRIPTION
**Pull request by TanyaAdams1**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -1,31 +1,30 @@
...
 
 parser = argparse.ArgumentParser(description='Crystal Graph Convolutional Neural Networks')
 parser.add_argument('data_options', metavar='OPTIONS', nargs='+',
                     help='dataset options, started with the path to root dir, '
-                    'then other options')
+                         'then other options')
 parser.add_argument('--task', choices=['regression', 'classification'],
-                    default='regression', help='complete a regression or '
-                    'classification task (default: regression)')
+                    default='classification', help='complete a regression or '
```
Comment (txie-93) [inline main.py]: Why do you change the default to classification?

File: main.py [hunk 2]
```diff
@@ -205,27 +219,27 @@ def train(train_loader, model, criterion, optimizer, epoch, normalizer):
     model.train()
 
     end = time.time()
-    for i, (input, target, _) in enumerate(train_loader):
+    for i, (_input, target, _) in enumerate(train_loader):
```
Comment (txie-93) [inline main.py]: Change the `_input` back to `input` for naming conventions.

File: main.py [hunk 3]
```diff
@@ -50,12 +49,19 @@
                     metavar='N', help='print frequency (default: 10)')
 parser.add_argument('--resume', default='', type=str, metavar='PATH',
                     help='path to latest checkpoint (default: none)')
-parser.add_argument('--train-size', default=None, type=int, metavar='N',
+parser.add_argument('--train-ratio', default=None, type=float, metavar='N',
                     help='number of training data to be loaded (default none)')
-parser.add_argument('--val-size', default=1000, type=int, metavar='N',
+parser.add_argument('--val-ratio', default=0.1, type=float, metavar='N',
```
Comment (txie-93) [inline main.py]: Can you try to capture the error when `--val-size` and `--val-ratio` are both used?

Comment (txie-93) [reply main.py]: Also, please change the document in README for the new interfaces.

File: main.py (hunk 1 above)
Comment (TanyaAdams1) [reply main.py]: It's just for debugging. If you think that the default should be regression, I'll change it back.

File: main.py (hunk 2 above)
Comment (TanyaAdams1) [reply main.py]: According to my linter, the name "input" shades the builtin "input", which is used for reading input from keyboard, therefore it suggests the change of the name.

File: main.py (hunk 3 above)
Comment (TanyaAdams1) [reply main.py]: Currently the behavior is that when both are used, the --val-size will override. I'll add an error capture if you think necessary.

I'll modify README, but whether to emit an error or to let one override when both options are present must be decided first.

File: main.py (hunk 1 above)
Comment (txie-93) [reply main.py]: Yes, the default should be regression because most people will use it. Please change it back.

File: main.py (hunk 2 above)
Comment (txie-93) [reply main.py]: I understand your concern. But `_input` usually means internal variables which is not appropriate in this case. Let's keep the name `input` for readability.

File: main.py (hunk 3 above)
Comment (txie-93) [reply main.py]: Let's explicitly capture this error when both are used. This will avoid confusions to the user.

**Comment by txie-93** [comment]
One more comment: could you update the README file to show that the code currently works for the latest version of the PyTorch?

**Comment by TanyaAdams1** [comment]
You mean to update the dependencies section, to set the pytorch version to the latest?

**Comment by txie-93** [comment]
Correct. Can you remove all the sentences that indicates the code only works up to v0.4?

**Comment by TanyaAdams1** [comment]
Updated.

**Comment by txie-93** [comment]
Excellent! Thanks for all the updates.

**Pull request merged**